server/src/main/java/com/microsoft/java/bs/core/internal/services/LifecycleService.java

```diff
@@ -155,6 +164,13 @@ private void updateGradleJavaHomeIfNecessary(URI rootUri) {
         File jdkInstallation = getJdkToLaunchDaemon(preferences.getJdks(), highestJavaVersion);
         if (jdkInstallation != null) {
           preferences.setGradleJavaHome(jdkInstallation.getAbsolutePath());
+        } else {
+          new ClientNotifier(client).sendNotification(
```

Comment:
Is it possible to expand the compatibility check scope?

Some cases for consideration:

1. If user specifies Gradle Java Home in preference, is it a valid Java home? Is it compatible with the gradle version being used?
2. If user does not specify the gradle version, we still can infer the gradle version used for the project, can we do the compatibility check?

Reply:
Commit [8731678](https://github.com/microsoft/build-server-for-gradle/pull/165/commits/87316784aaf454d061a832bec0a38694301f8149) address this. We are now checking for `GradleJavaHome`, if it is available and compatible. If not we move to find a compatible JDK from the list of JDKs we get from preferences.